shipgen/MuonBackGenerator.cxx

```diff
@@ -183,11 +184,13 @@ Bool_t MuonBackGenerator::ReadEvent(FairPrimaryGenerator* cpg)
          px = track->GetPx();
          py = track->GetPy();
          pz = track->GetPz();
-         if (fPaintBeam > 0) {
+         if (fPhiRandomize) {
+             gRandom->SetSeed(12345);  // TO BE FIXED: this should not be hardcoded
+             double phi_rm = gRandom->Uniform(0., 2*TMath::Pi());
              Double_t phi0 = TMath::ATan2(py, px);
              Double_t pt = track->GetPt();
-             px = pt * TMath::Cos(phi + phi0);
-             py = pt * TMath::Sin(phi + phi0);
+             px = pt * TMath::Cos(phi_rm + phi0);
```

Comment:
Probably no need to calculate phi0 or performing the addition, as the distribution of (uniform(0, 2pi) + const) % 2pi should be the same as without the constant.